### PR TITLE
Jvenstad/zone cleanup 3

### PR DESCRIPTION
**Pull request by jonmv**
@andreer please review. 

Must be merged with https://git.ouroath.com/vespa/hosted/pull/5111

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/versions/VersionStatus.java [hunk 1]
```diff
@@ -120,8 +120,9 @@ public static VersionStatus compute(Controller controller, Version currentVersio
 
     private static ListMap<Version, String> findConfigServerVersions(Controller controller) {
         List<URI> configServers = controller.zoneRegistry().zones().stream()
+                // TODO: Filter properly.
                 .filter(zone -> ! zone.region().equals(RegionName.from("us-east-2a")))
-                .flatMap(zone -> controller.getConfigServerUris(zone.environment(), zone.region()).stream())
+                .flatMap(zone -> controller.getConfigServerUris(zone).stream())
```
Comment (bjorncs) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/versions/VersionStatus.java]: Consider renaming the lambda variable `zone` to `zoneId`.

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/identifiers/DeploymentId.java [hunk 2]
```diff
@@ -3,55 +3,52 @@
 
 import com.yahoo.config.provision.ZoneId;
 
+import java.util.Objects;
+
 /**
- * Application + zone. 
+ * ApplicationId x ZoneId.
```
Comment (bjorncs) [inline controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/identifiers/DeploymentId.java]: Should _x_ be _+_ ?

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/ZoneRegistryMock.java [hunk 3]
```diff
@@ -81,18 +82,33 @@ public boolean hasZone(ZoneId zoneId) {
...
+        URI kibanaPath = URI.create(kibanaQuery);
+        return Optional.of(URI.create(String.format("http://log.%s.test", deploymentId.zoneId().value())).resolve(kibanaPath));
     }
 
     @Override
     public Duration getDeploymentTimeToLive(ZoneId zoneId) {
-        return deploymentTimeToLive.get(zoneId);
+        return deploymentTimeToLive.containsKey(zoneId)
+                ? deploymentTimeToLive.get(zoneId)
+                : Duration.ofMillis(Long.MAX_VALUE / 2);
```
Comment (bjorncs) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/ZoneRegistryMock.java]: The default TTL value is hard to understand. I would not interpret it as infinite, but as a specific default value. `Optional<Duration>` seems more obvious.

**Review by bjorncs — CHANGES_REQUESTED** [review]
See comments. Overall 👍 👍 👍

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/identifiers/DeploymentId.java (hunk 2 above)
Comment (jonmv) [reply controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/identifiers/DeploymentId.java]: Not if you ask me :p It is the [cartesian product](https://en.wikipedia.org/wiki/Cartesian_product) of the two.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/versions/VersionStatus.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/versions/VersionStatus.java]: Yes, there may be a lot more of these lying around. I didn't prioritise looking for them, but I will take a look.

I'll rename this in my next PR.

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/ZoneRegistryMock.java (hunk 3 above)
Comment (mpolden) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/ZoneRegistryMock.java]: Agree. Why change the signature? Most deployments do not have a TTL, so optional makes more sense.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/zone/v1/ZoneApiHandler.java [hunk 4]
```diff
@@ -109,9 +109,7 @@ private HttpResponse environment(HttpRequest request, Environment environment) {
 
     private HttpResponse defaultRegion(HttpRequest request, Environment environment) {
         RegionName region = zoneRegistry.getDefaultRegion(environment)
-                                        .orElseThrow(() -> new IllegalArgumentException(
-                                                "No default region for environment: " + environment
-                                        ));
+                .orElseThrow(() -> new IllegalArgumentException("No default region for environment: " + environment));
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/zone/v1/ZoneApiHandler.java]: Breaks indentation.

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/zone/v1/ZoneApiHandler.java]: Thanks, I'll fix in next PR.

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/zone/v1/ZoneApiHandler.java]: What setting? This is IntelliJ for me.

**Review by bjorncs — COMMENTED** [review]
👍

**Pull request merged**